src/master_mapping.rs

```diff
@@ -0,0 +1,247 @@
+use super::decode_data::*;
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct MessageInfo {
+    pub description: String,
+    pub decoder: fn(data: &[u8]) -> HashMap<u8, f32>,
+}
+
+impl MessageInfo {
+    pub fn new(description: String, decoder: fn(data: &[u8]) -> HashMap<u8, f32>) -> Self {
+        Self {
+            description,
+            decoder,
+        }
+    }
+}
+
+// Mapping from external message ID to decoding information
+pub fn get_message_info(id: &u32) -> MessageInfo {
+    match id {
+        1 => return MessageInfo::new("accumulator status".to_string(), decode_accumulator_status),
```

Comment:
I know there probably isnt a cleaner way of doing this but I wanna voice that this feels gross haha

Reply:
😬 yeahhhhhhh